timezone_manager.py

```
@@ -265,28 +268,35 @@ async def request_tz_task() -> str:
 
             async with aiohttp.ClientSession() as session:
                 while True:  # Retry loop
-                    try:
-                        await asyncio.sleep(1)
-                        async with session.get(TZ_GETTER_URL) as response:
-                            if response.status == 200:  # Break on success
-                                str_content = await response.text()
-                                tz = json.loads(str_content).get("tz")
-                                if tz is not None:
-                                    return TimezoneManager.update_timezone(tz)
-                                return "Invalid response from server"
-                            else:
-                                logger.warning(
-                                    f"timezone fetch failed with status code: {response.status}, re-fetching"
-                                )
-                    except aiohttp.ClientError as e:
-                        logger.warning(f"Request failed: {e}. Retrying...")
+                    await asyncio.sleep(1)
+                    async with session.get(TZ_GETTER_URL) as response:
+                        if response.status == 200:  # Break on success
+                            str_content = await response.text()
+                            tz = json.loads(str_content).get("tz")
+                            if tz is not None:
+                                TimezoneManager.update_timezone(
+                                    tz
+                                )  # raises TimezoneManagerError if fails
+                                return tz
+                            logger.warning("Invalid response from server, re-fetching")
+                        else:
+                            logger.warning(
+                                f"timezone fetch failed with status code: {response.status}, re-fetching"
+                            )
```

This should probably also raise an exception?

---

I didn't make it raise an exception as I wanted the exceptions to break the while loop, in this case I want it to re-try fetching the data

---

Ohh, yeah, makes sense. Sorry